paper-api/src/main/java/org/bukkit/OfflinePlayer.java

```diff
@@ -290,13 +291,26 @@ public default BanEntry banPlayer(@Nullable String reason, @Nullable java.util.D
     // Paper end
 
     /**
-     * Gets the Location where the player will spawn at, null if they
+     * Gets the Location where the player will spawn at, {@code null} if they
+     * don't have a valid respawn point.
+     *
+     * @return respawn location if exists, otherwise {@code null}.
+     */
+    @Nullable
+    default Location getRespawnLocation() {
+        return this.getRespawnLocation(false); // keep old behavior for offline players
+    }
```

Comment:
Thoughts on deprecating this method (without overload) altogether?
`Player` extends `OfflinePlayer`, and both have different default chunk loading behavior, which is quite confusing (also would be nice to specify the default behavior in respective javadocs).

Reply:
I have updated the javadoc, not sure for the deprecation, i kinda expect offline player data to be less accurate than online by default. What i think is confusing is the fact that an online player extends an offline player in the api.

Reply:
> What i think is confusing is the fact that an online player extends an offline player in the api.

Yes, that was my concern. Technically, you can have `OfflinePlayer` object representing an online `Player` instance (`#getOfflinePlayer` too returns online player if present), so you may call the method on an offline player expecting not to load chunks, but they will load due to the object being an online player.

New devs might stumble upon this quirk and will have to account for it. Deprecation would require to explicitly specify the wanted behavior, but I don't have a strong feeling against leaving this as is.